app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: tuple) -> str:
+    archive = {}
+
+    def inner(*args, **kwargs) -> Callable:
+        if args not in archive:
+            result = func(*args, **kwargs)
+            archive[args] = result
+            print("Calculating new result")
+        else:
+            print("Getting from cache")
+        return archive[args]
+    return inner
```

Comment:
Regarding type annotation, the previous comment was not implemented in a full manner. `cache` function should return `Callable` because we will return the `inner` function, this is the only option, not `str`. At the same time, the `inner` function should [return](https://docs.python.org/3/library/typing.html#typing.Any) `Any` from `typing`, because the result may differ depending on the input and we cannot foresee all possible options.